src/main/java/cn/nukkit/entity/item/EntityBoat.java

```diff
@@ -409,6 +410,12 @@ public boolean canPassThrough() {
     public void kill() {
         super.kill();
 
+        if (this.lastDamageCause instanceof EntityDamageByEntityEvent) {
+            Entity damager = ((EntityDamageByEntityEvent) this.lastDamageCause).getDamager();
+            if (damager instanceof Player && ((Player) damager).isCreative()) {
+                return;
+            }
+        }
```

Comment:
what is this for?

Reply:
vehicles broken in creative mode should disappear